keymapper/injection/keycode_mapper.py

```diff
@@ -236,9 +237,23 @@ def __init__(self, context, source, forward_to):
 
     def macro_write(self, ev_type, code, value):
```

Comment:
how about an extra parameter to `macro_write`?

```suggestion
    def macro_write(self, ev_type, code, value, to_device=False):
        if to_device:
            ...
```

and then in macros.py in `to_device` (or `led`, whatever):

```py
self.tasks.append(lambda handler: handler(ev_type, code, value, to_device=True))
```